core/dbt/parser/manifest.py

```diff
@@ -1439,6 +1440,63 @@ def _process_refs(
         node.depends_on.add_node(target_model_id)
 
 
+def _process_metric_node(
+    manifest: Manifest,
+    current_project: str,
+    metric: Metric,
+) -> None:
+    """Sets a metric's input_measures"""
+
+    # This ensures that if this metrics input_measures have already been set
+    # we skip the work. This could happen either due to recursion or if multiple
+    # metrics derive from another given metric.
+    # NOTE: This does not protect against infinite loops
+    if len(metric.type_params.input_measures) > 0:
+        return
+
+    if metric.type is MetricType.SIMPLE or metric.type is MetricType.CUMULATIVE:
```

Comment:
Typically for an if statement like this I'd do more of a
```python
if metric.type in [MetricType.SIMPLE, MetricType.CUMULATIVE]:
...
```
However the type checker help of `assert_values_exhausted` doesn't work with that kind of formatting. Thus the individual `is` based checked.